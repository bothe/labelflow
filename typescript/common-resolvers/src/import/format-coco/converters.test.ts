import {
  isCocoSegmentationBox,
  convertGeometryFromCocoAnnotationToLabel,
} from "./converters";

describe(isCocoSegmentationBox, () => {
  it("spots segmentation box", () => {
    expect(
      isCocoSegmentationBox([
        [
          123.33, 254.0001, 369.12346, 254.0005, 369.12348, 567.09436, 123.33,
          567.09456, 123.33, 254.0001,
        ],
      ])
    ).toBeTruthy();
  });

  it("considers an empty segmentation to a box, based on the bounding box", () => {
    expect(isCocoSegmentationBox([])).toBeTruthy();
  });

  it("discards segmentation with more than one polygon", () => {
    expect(
      isCocoSegmentationBox([
        [
          123.33, 254.0001, 369.12346, 254.0005, 369.12348, 567.09436, 123.33,
          567.09456, 123.33, 254.0001,
        ],
        [100, 0],
      ])
    ).toBeFalsy();
  });

  it("discards segmentation with more than ten coordinates", () => {
    expect(
      isCocoSegmentationBox([
        [
          123.33, 254.0001, 369.12346, 254.0005, 369.12348, 567.09436, 123.33,
          567.09456, 123.33, 254.0001, 100, 0,
        ],
      ])
    ).toBeFalsy();
  });

  it("discards reordered segmentation of a box", () => {
    expect(
      isCocoSegmentationBox([
        [
          123.33, 254.0001, 369.12346, 254.0005, 369.12348, 567.09436, 123.33,
          567.09456, 254.0001, 123.33,
        ],
      ])
    ).toBeFalsy();
  });

  it("discards degenerated box (reordered points)", () => {
    expect(
      isCocoSegmentationBox([
        [
          123.33, 254.0001, 369.12348, 567.09436, 369.12346, 254.0005, 123.33,
          567.09456, 123.33, 254.0001,
        ],
      ])
    ).toBeFalsy();
  });
});

describe(convertGeometryFromCocoAnnotationToLabel, () => {
  it("converts a box into a box", () => {
    expect(
      convertGeometryFromCocoAnnotationToLabel(
        [
          [
            123.33, 254.0001, 369.12346, 254.0005, 369.12348, 567.09436, 123.33,
            567.09456, 123.33, 254.0001,
          ],
        ],
        [0, 0, 568, 568],
        568
      )
    ).toMatchSnapshot();
  });

  it("converts an empty segmentation's bounding box into a box", () => {
    expect(
      convertGeometryFromCocoAnnotationToLabel([], [100, 200, 300, 400], 700)
    ).toMatchSnapshot();
  });

  it("converts a polygon into a polygon", () => {
    expect(
      convertGeometryFromCocoAnnotationToLabel(
        [
          [
            123.33, 254.0001, 369.12346, 254.0005, 369.12348, 567.09436, 123.33,
            567.09456, 100, 550, 123.33, 254.0001,
          ],
        ],
        [0, 0, 568, 568],
        568
      )
    ).toMatchSnapshot();
  });
});
