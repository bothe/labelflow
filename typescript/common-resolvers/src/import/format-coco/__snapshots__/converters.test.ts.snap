// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertGeometryFromCocoAnnotationToLabel converts a box into a box 1`] = `
Object {
  "geometry": Object {
    "coordinates": Array [
      Array [
        Array [
          123.33,
          313.9999,
        ],
        Array [
          369.12346,
          313.9995,
        ],
        Array [
          369.12348,
          0.9056399999999485,
        ],
        Array [
          123.33,
          0.9054399999999987,
        ],
        Array [
          123.33,
          313.9999,
        ],
      ],
    ],
    "type": "Polygon",
  },
  "type": "Box",
}
`;

exports[`convertGeometryFromCocoAnnotationToLabel converts a polygon into a polygon 1`] = `
Object {
  "geometry": Object {
    "coordinates": Array [
      Array [
        Array [
          123.33,
          313.9999,
        ],
        Array [
          369.12346,
          313.9995,
        ],
        Array [
          369.12348,
          0.9056399999999485,
        ],
        Array [
          123.33,
          0.9054399999999987,
        ],
        Array [
          100,
          18,
        ],
        Array [
          123.33,
          313.9999,
        ],
      ],
    ],
    "type": "Polygon",
  },
  "type": "Polygon",
}
`;

exports[`convertGeometryFromCocoAnnotationToLabel converts an empty segmentation's bounding box into a box 1`] = `
Object {
  "geometry": Object {
    "coordinates": Array [
      Array [
        Array [
          100,
          600,
        ],
        Array [
          100,
          200,
        ],
        Array [
          400,
          200,
        ],
        Array [
          400,
          600,
        ],
        Array [
          100,
          600,
        ],
      ],
    ],
    "type": "Polygon",
  },
  "type": "Box",
}
`;
